README.md

```diff
@@ -27,16 +36,72 @@ represents the "RGB vector space" version.
 
 ![ColorVectorSpace](images/comparison.png "Comparison")
 
+This package also defines `norm(c)` for RGB and grayscale colors.
+This makes these color spaces [normed vector spaces](https://en.wikipedia.org/wiki/Normed_vector_space).
+Note that `norm` has been designed to satisfy equivalence of grayscale and RGB representations: if
+`x` is a scalar, then `norm(x) == norm(Gray(x)) == norm(RGB(x, x, x))`.
+
 ## Usage
 
 ```jl
 using ColorTypes, ColorVectorSpace
 ```
 
-That's it. Just by loading `ColorVectorSpace`, most basic mathematical
+For the most part, that's it; just by loading `ColorVectorSpace`, most basic mathematical
 operations will "just work" on `AbstractRGB`, `AbstractGray`
 (`OpaqueColor{T,1}`), `TransparentRGB`, and `TransparentGray` objects.
 (See definitions for the latter inside of `ColorTypes`).
 
-If you discover missing operations, please open an issue, or better
-yet submit a pull request.
+However, there are some additional operations.
+
+### Multiplication
+
+Grayscale values are conceptually similar to scalars, and consequently it seems straightforward to define multiplication of two grayscale values.
+RGB values present more options.
+This package supports three different notions of multiplication: the inner product, the hadamard (elementwise) product, and the tensor product.
+
+```julia
+julia> c1, c2 = RGB(0.2, 0.3, 0.4), RGB(0.5, 0.3, 0.2)
+(RGB{Float64}(0.2,0.3,0.4), RGB{Float64}(0.5,0.3,0.1))
+
+julia> c1⋅c2     # or dot(c1, c2)
+0.09000000000000001
+
+julia> c1⊙c2     # or hadamard(c1, c2)
+RGB{Float64}(0.1,0.09,0.08000000000000002)
+
+julia> c1⊗c2     # or tensor(c1, c2)
+RGBRGB{Float64}(
+ 0.1   0.06  0.04000000000000001
+ 0.15  0.09  0.06
+ 0.2   0.12  0.08000000000000002)
+```
+
+Note that `c1⋅c2 = (c1.r*c2.r + c1.g*c2.g + c1.b*c2.b)/3`, where the division by 3 ensures the equivalence of `x`, `Gray(x)`, and `RGB(x, x, x)` for scalar `x` as described above for `norm`.
+
+It is not obvious that one of these should be the default notion of multiplication, so the ordinary multiplication operation `*` is undefined for RGB colors.
+(It is defined for grayscale.)
+
+### Variance
+
+The variance `v = E((c - μ)^2)` (or its bias-corrected version) involves a multiplication,
+and consistent with the above you must specify which sense of multiplication you wish to use:
```

Comment:
I'm not sure I know anymore how it works in other languages :smile:. In Matlab, I guess `var(img)` would include the variance over color channels and `var(img, [1, 2])` would compute variance within color channels? Is the former even useful?

Reply:
It's just for clarification purposes only, because these are not available in other languages. I think it's more common to use the channelwise version `[var(img[:, :, 1]), var(img[:, :, 2]), var(img[:, :, 3])]` in practice, which is equivalent to `varmult(⊙, img)`.